packages/web-app-files/tests/unit/components/SideBar/Exif/ExifPanel.spec.ts

```diff
@@ -0,0 +1,140 @@
+import ExifPanel from '../../../../../src/components/SideBar/Exif/ExifPanel.vue'
+import { defaultPlugins, shallowMount } from 'web-test-helpers'
+import { mock } from 'vitest-mock-extended'
+import { Photo, Image, GeoCoordinates } from '@ownclouders/web-client/src/graph/generated'
+import { Resource } from '@ownclouders/web-client'
+
+describe('Exif SideBar Panel', () => {
+  describe('photo metadata', () => {
+    const keys = [
+      'cameraMake',
+      'cameraModel',
+      'focalLength',
+      'fNumber',
+      'exposureTime',
+      'iso',
+      'orientation',
+      'takenDateTime'
+    ]
+    it.each(keys)('shows value in panel for key "%s"', (key) => {
+      const resource = mock<Resource>({
+        photo: mock<Photo>({
+          cameraMake: 'Canon',
+          cameraModel: 'Canon EOS 1300D',
+          focalLength: 222,
+          fNumber: 5.6,
+          exposureNumerator: 1,
+          exposureDenominator: 20,
+          iso: 320,
+          orientation: 1,
+          takenDateTime: '2017-02-11T14:54:50Z'
+        })
+      })
+      const { wrapper } = createWrapper({ resource })
+      expect(wrapper.find(`[data-testid="exif-panel-${key}"]`).text()).toMatchSnapshot()
```

Comment:
Isn't it a bit weird to have a snapshot for a single value?

Reply:
Hm... actually, what's your take on that? I made it a snapshot test because most of the values get some string modification for being displayed. E.g. if the `fNumber` value is `16.6` it will be rendered as `16.6 mm`. I didn't want to rebuild the string modification logic in the tests again, so that's why I decided for snapshot tests. And because I hate snapshot tests I made the scope as small as possible (only the value, no surrounding html).

Reply:
Ah, fair enough. 👍🏻